src/Network/Email/AbstractTransport.php

```diff
@@ -1,3 +1,16 @@
 <?php
-// @deprecated Backward compatibility with 2.x, 3.0.x
-class_alias('Cake\Mailer\AbstractTransport', 'Cake\Network\Email\AbstractTransport');
+// @codingStandardsIgnoreFile
+namespace Cake\Network\Email;
+
+class_alias(\Cake\Mailer\AbstractTransport::class, AbstractTransport::class);
+
+if (class_exists(AbstractTransport::class)) {
+    return;
+}
```

Comment:
@thinkingmedia Does this code really work without any warnings? We can use classes before declaring, if they don't implement any interfaces and don't use any traits. For example, the following is a valid code:

```php
$d = new Derived;
$d->foo();

return;

class Base {}

class Derived extends Base {
	function foo() {
		echo __METHOD__;
	}
}
```

As an alternative, `__halt_compiler()` might work if PHPStorm doesn't support it. But isn't there any way to satisfy IDEs without such tricks?